app/controllers/inventories_controller.rb

```diff
@@ -67,18 +59,42 @@ def model_params
     params.require(:model_id)
   end
 
+  def set_data
+    @inventory = Inventory.new
+    @models = Model.kept.select(:id, :name).order(:name)
+    @body_colors = BodyColor.select(:id, :name).to_json.html_safe
+    @memories = Memory.select(:id, :display_name).to_json.html_safe
+  end
+
   def convert_params
     arr = []
-    params[:model][:name].each do |key, value|
-      item = {}
+    params[:model].each do |key, value|
+      os_version = validated_os_version(params[:os_version][key])
+      return false unless os_version
 
-      item[:model] = Model.where(name: value.split(' (')[0]).first
-      item[:body_color] = BodyColor.where(name: params[:body_color][:name][key]).first_or_initialize
-      item[:memory] = Memory.where(amount: params[:memory][:amount][key], display_name: params[:memory][:display_name][key]).first_or_initialize
-      item[:os_version] = OsVersion.where(major: params[:os_version][:major][key], minor: params[:os_version][:minor][key], patch: params[:os_version][:patch][key]).first_or_initialize
-      item[:inventory] = Inventory.new(quantity: params[:inventory][:quantity][key], price: params[:inventory][:price][key])
+      item = {}
+      item[:model] = Model.find(value)
+      item[:body_color] = BodyColor.find(params[:body_color][key])
+      item[:memory] = Memory.find(params[:body_color][key])
+      item[:os_version] = OsVersion.where(
+        major: os_version[0],
+        minor: os_version[1].nil? ? 0 : os_version[1],
+        patch: os_version[2].nil? ? 0 : os_version[2],
```

Comment:
Default value for any table's column should be common for most of the case.
Then, it should be defined in model class. like this way.
https://stackoverflow.com/a/4576026

Reply:
In this case, I'm using select query with where so I'm not sure about understanding your comment.
BTW, about default value, I set 0 is a default value of minor and patch in database (through migration files). Is it ok or we should move the default value in model?

Reply:
> I set 0 is a default value of minor and patch in database (through migration files)

I usually define only database's default value or both of database and application.
Defining database's default value is more certain for keeping data consistency because when application is transferred another application but database keep being used, the default value is also same.

When the object is only initialized, database's default value isn't set. Then, defining default value on application is required.